sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -1067,4 +1068,142 @@ public HttpPipeline getHttpPipeline() {
                 return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), null, twinHeaders));
             });
     }
+
+    /**
+     * Query digital twins.
+     * @param query The query string, in SQL-like syntax.
+     * @return A {@link PagedFlux} of application/json query result items.`
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> query(String query) {
+        return new PagedFlux<>(
+            () -> withContext(context -> queryFirstPage(query, context)),
+            nextLink -> withContext(context -> queryNextPage(query, nextLink, context)));
```

Comment:
does `queryNextPage` need the query string again?

Reply:
It does not need it but no harm in passing it as well. To keep it consistent with .net, I will remove it here.